app/models/user.rb

```diff
@@ -21,12 +21,27 @@ def self.dashboard_headers
     DASHBOARD_ATTRS + Training::Module.live.map { |mod| "module_#{mod.position}_time" }
   end
 
+  # @param email [String]
+  # @param gov_one_id [String]
+  # @return [User]
+  def self.find_or_create_from_gov_one(email:, gov_one_id:)
```

Comment:
This doesn't yet have a unit test. Now all the post login logic is here the `User` object can be tested more easily.